applications/GeoMechanicsApplication/tests/cpp_tests/test_transport_equation_utilities.cpp

```diff
@@ -0,0 +1,63 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "custom_utilities/transport_equation_utilities.hpp"
+#include "testing/testing.h"
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(CalculateBiotModulusInverse_GivesExpectedResult, KratosGeoMechanicsFastSuite)
+{
+    Properties properties;
+    properties[IGNORE_UNDRAINED]          = false;
+    properties[POROSITY]                  = 0.5;
+    properties[BULK_MODULUS_SOLID]        = 1.0e9;
+    properties[BULK_MODULUS_FLUID]        = 2.0e6;
+    const double biot_coefficient         = 1.0;
+    const double degree_of_saturation     = 0.3;
+    const double derivative_of_saturation = 0.2;
+
+    const double expected_value = -0.09999992485;
+    KRATOS_EXPECT_DOUBLE_EQ(GeoTransportEquationUtilities::CalculateBiotModulusInverse(
+                                biot_coefficient, degree_of_saturation, derivative_of_saturation, properties),
+                            expected_value);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CalculateBiotModulusInverse_ReturnsLargeNumber_WhenIgnoreUndrained, KratosGeoMechanicsFastSuite)
+{
+    Properties properties;
+    properties[IGNORE_UNDRAINED]          = true;
+    properties[POROSITY]                  = 0.5;
+    properties[BULK_MODULUS_SOLID]        = 1.0e9;
+    properties[BULK_MODULUS_FLUID]        = 2.0e6;
+    const double biot_coefficient         = 1.0;
+    const double degree_of_saturation     = 0.3;
+    const double derivative_of_saturation = 0.2;
+
+    const auto large_number = 1e10;
+    KRATOS_EXPECT_TRUE(GeoTransportEquationUtilities::CalculateBiotModulusInverse(
+                           biot_coefficient, degree_of_saturation, derivative_of_saturation, properties) > large_number)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CalculateBiotModulusInverse_DoesNotThrow_ForEmptyProperties, KratosGeoMechanicsFastSuite)
```

Comment:
Is this desired behavior, since it would require me to test the returned value before actually using it ...?

Reply:
This question is exactly the reason I wanted to make it explicit in this test. I don't know exactly what the desired behavior is. E.g. do we want to always check if all the properties are set with physical values inside this function, or do we assume it's okay? It is user input though (via the MaterialParameters), so it could indeed result in NaN now if the wrong inputs are given.

I can of course add checks and throw (and UT for that), what's your opinion?

Reply:
I would suggest to not change the behavior yet, since that would be beyond the scope of this PR. However, I would like you to add an issue to the product backlog about it, so we can revisit it later. Now it has been documented through a unit test, which is a first step in the right direction, so thanks for that.